src/main/java/duke/commands/Command.java

```diff
@@ -0,0 +1,39 @@
+package duke.commands;
+
+import duke.ItemList;
+import duke.DukeException;
+import duke.Ui;
+
+/**
+ * Represents functionalities of commands that a <code>Duke</code> object can
+ * interpret and execute.
+ */
+public abstract class Command {
+    /**
+     * Executes the command by altering the inputs.
+     * 
+     * @param itemList List to update.
+     * @param ui UI to update.
+     * @throws DukeException If errors occur.
+     */
+    public abstract void execute(ItemList itemList, Ui ui) throws DukeException;

```

Comment:
Plural form should be used to represent Collections.
(same for the classes that extend Command)
```suggestion
    public abstract void execute(ItemList items, Ui ui) throws DukeException;

```